src/main/kotlin/io/snyk/plugin/ui/jcef/OpenFileLoadHandlerGenerator.kt

```diff
@@ -0,0 +1,70 @@
+package io.snyk.plugin.ui.jcef
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.ui.jcef.JBCefBrowser
+import com.intellij.ui.jcef.JBCefJSQuery
+import io.snyk.plugin.getDocument
+import io.snyk.plugin.navigateToSource
+import io.snyk.plugin.snykcode.core.SnykCodeFile
+import io.snyk.plugin.toVirtualFile
+import org.cef.browser.CefBrowser
+import org.cef.browser.CefFrame
+import org.cef.handler.CefLoadHandlerAdapter
+
+class OpenFileLoadHandlerGenerator(snykCodeFile: SnykCodeFile) {
+    private val project = snykCodeFile.project
+
+    fun openFile(value: String): JBCefJSQuery.Response {
+        var values = value.replace("\n", "").split(":")
+        var filePath = values[0]
+        var startLine = values[1].toInt()
+        var endLine = values[2].toInt()
+        var startCharacter = values[3].toInt()
+        var endCharacter = values[4].toInt()
+
+        ApplicationManager.getApplication().invokeLater {
+            val virtualFile = filePath.toVirtualFile()
```

Comment:
This will need my Windows PR you are reviewing (#503)

Reply:
Actually, I'd get the virtualFile from the SnykCodeFile. This avoids you executing long running code in the UI Thread.